tests/functional/metadata/mongodb/delObject.spec.js

```diff
@@ -0,0 +1,365 @@
+                // only 1 object remaining in db
+                getObjectCount((err, count) => {
+                    assert.deepStrictEqual(err, null);
+                    assert.strictEqual(count, 1);
+                    return next();
+                });
+            },
+        ], done);
+    });
+
+    it('Should not throw error when object non existent', done => {
+        const objName = 'non-existent-object';
+        const bucketName = BUCKET_NAME;
+        return metadata.deleteObjectMD(bucketName, objName, null, logger, err => {
+            assert.deepStrictEqual(err, null);
+            return done();
+        });
+    });
+
+    it('Should not throw error when bucket non existent', done => {
+        const objName = 'non-existent-object';
+        const bucketName = BUCKET_NAME;
+        return metadata.deleteObjectMD(bucketName, objName, null, logger, err => {
+            assert.deepStrictEqual(err, null);
+            return done();
+        });
+    });
+
+    it('Master should not be updated when non lastest version is deleted', done => {
+        const bucketName = BUCKET_NAME;
+        let versionId1 = null;
+        const params = {
+            objName: 'test-object',
+            objVal: {
+                key: 'test-object',
+                versionId: 'null',
+            },
+            vFormat: 'v0',
+        };
+        const versionParams = {
+            versioning: true,
+            versionId: null,
+            repairMaster: null,
+        };
+        return async.waterfall([
+            next => {
+                // we start by creating a new version and master
+                versionId1 = generateVersionId(this.replicationGroupId);
+                params.versionId = versionId1;
+                params.objVal.versionId = versionId1;
+                versionParams.versionId = versionId1;
+                metadata.putObjectMD(BUCKET_NAME, params.objName, params.objVal,
+                versionParams, logger, err => {
+                    if (err) {
+                        return next(err);
+                    }
+                    return next();
+                });
+            },
+            next => {
+                // we create a second version of the same object (master is updated)
+                params.objVal.versionId = 'version2';
+                versionParams.versionId = null;
+                metadata.putObjectMD(BUCKET_NAME, params.objName, params.objVal,
+                versionParams, logger, err => {
+                    if (err) {
+                        return next(err);
+                    }
+                    return next();
+                });
+            },
+            next => {
+                // we delete the first version
+                metadata.deleteObjectMD(bucketName, params.objName, { versionId: versionId1 },
+                logger, err => {
+                    assert.deepStrictEqual(err, null);
+                    return next();
+                });
```

Comment:
just pass `next` here, and move `err` assert to the end of waterfall (where `done` is passed)... and actually mocha already handles this, so if an error is passed to `done` then the test will fail... So this is very simple actually:

```suggestion
                    logger, next);
```

Reply:
fixed in https://github.com/scality/Arsenal/pull/1729/commits/a6f48b9f342344e6df5ca2e3b7a2d03d8737fbba